WalletWasabi.Fluent/ViewModels/Login/LoginViewModel.cs

```diff
@@ -15,7 +17,7 @@ public partial class LoginViewModel : RoutableViewModel
 		[AutoNotify] private bool _isPasswordNeeded;
 		[AutoNotify] private string _walletName;
 
-		public LoginViewModel(WalletViewModelBase walletViewModelBase)
+		public LoginViewModel(WalletViewModelBase walletViewModelBase, LegalChecker legalChecker)
```

Comment:
At this point legalChecker is like a virus. Isn't any other way to have a reference this checker? I didn't think much about this yet but it smells weird to me.

Reply:
This is the current UI concept, there is no other way. Currently, they are creating a new ViewModel for different wallet states and for different wallet types ~7 files.

Reply:
`if (legalChecker.TryGetNewLegalDocs(out ...)` is there several times, so Lucas may be right. 

I can see:

* LoginViewModel.cs: `if (legalChecker.TryGetNewLegalDocs(out _)) { ... new TermsAndConditionsViewModel(legalChecker, targetViewModel) ... }`
* TermsAndConditionsViewModel.cs: `if (legalChecker.TryGetNewLegalDocs(out ...`

Is it necessary to check `TryGetNewLegalDocs` several times?